src/umpire/device_allocator_helper.cpp

```diff
@@ -64,13 +65,15 @@ __host__ __device__ DeviceAllocator get_device_allocator(const char* name)
 
 __host__ __device__ DeviceAllocator get_device_allocator(int id)
 {
-  if (id < 0 || id > UMPIRE_TOTAL_DEV_ALLOCS) {
+  if (id < 1 || id > UMPIRE_TOTAL_DEV_ALLOCS) {
```

Comment:
This does not seem to be correct.  The user should be given a negative number as an ID.  So, when they call get_device_allocator() it will be a negative number that they pass in right?

Reply:
The id needs to be converted here, at the very top of the function.